### PR TITLE
test: add unit/integration tests for configs and recipes

### LINKED ISSUE #198: [MINOR] §5: No unit or integration tests for justfile recipes or configs
## Finding
**Severity:** MINOR
**Section:** 5
**Evidence:** repo/
**Principle:** TDD

Beyond e2e tests (which require live infrastructure), there are no unit or integration tests for the justfile recipes, NATS config syntax, Nomad HCL syntax, or docker-compose validity. `just validate-configs` exists but only syntax-checks HCL — it does not validate NATS config or compose files.

---
Part of #174

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Implements unit and integration tests for justfile recipes, NATS config syntax validation, Nomad HCL validation, and docker-compose file validity. Extends `just validate-configs` to cover NATS and Compose validation in addition to existing HCL checks.

## Changes
- Add `scripts/validate_nats_config.py` — validates NATS server and leaf configuration syntax
- Add `scripts/validate_compose.py` — validates docker-compose file structure and image references
- Add `tests/test-config-validators.sh` — integration tests for NATS and Compose validators
- Add `tests/test-justfile-recipes.sh` — unit tests for justfile recipe definitions
- Update `.github/workflows/ci.yml` — run new config and recipe tests in CI pipeline
- Update `justfile` — integrate new validators into `just validate-configs`
- Remove obsolete validation scripts — `tools/validate-nats-auth.sh`, `scripts/check-doc-field-drift.sh`, `scripts/check_e2e_image_pins.sh`

## Testing
- Run `just validate-configs` to verify NATS, Compose, and HCL validation
- Run `tests/test-config-validators.sh` to execute config validator tests
- Run `tests/test-justfile-recipes.sh` to execute justfile recipe tests
- Verify CI workflow includes and passes new test steps

## Closes
Closes #198

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: .github/workflows/_required.yml [hunk 1]
```diff
@@ -203,19 +203,22 @@ jobs:
         with:
           submodules: true
 
-      - name: Validate NATS config syntax
+      - name: Install just
+        uses: extractions/setup-just@53165ef7e734c5c07cb06b3c8e7b647c5aa16db3  # v4
+
+      - name: Install PyYAML (for compose validator)
+        run: pip install pyyaml
```
Comment (mvillmow) [inline .github/workflows/_required.yml]: `pip install pyyaml` runs without an explicit `actions/setup-python` step, so it targets the runner's system Python. This works today on `ubuntu-latest` (python3+pip preinstalled, and the `build` job at line 277 uses the same bare `pip install` pattern), but a future runner image with an externally-managed system Python (PEP 668) could break it with `error: externally-managed-environment`. Consider adding `actions/setup-python` before the pip installs for both jobs, or `pip install --user pyyaml`, to insulate from runner-image changes. Non-blocking — current behavior is correct.

**Review by mvillmow — COMMENTED** [review]
GO: closes #198's 4 gaps with binary-free validators that run in CI; exit_code symbol + non-vacuous gates verified against actual files. One minor: bare pip install w/o setup-python.

File: .github/workflows/_required.yml [hunk 2]
```diff
@@ -203,19 +203,22 @@ jobs:
         with:
           submodules: true
 
-      - name: Validate NATS config syntax
+      - name: Install just
+        uses: extractions/setup-just@53165ef7e734c5c07cb06b3c8e7b647c5aa16db3  # v4
+
+      - name: Install PyYAML (for compose validator)
+        run: pip install --user pyyaml
```
Comment (mvillmow) [inline .github/workflows/_required.yml:213]: Re-opening: prior review comment not addressed — Validate job switched to `pip install --user pyyaml`, but the comment also named the `build` job at line 277, which still runs bare `pip install yamllint` against system Python (_required.yml:277). The 'both jobs' concern is only half-resolved.

> `pip install pyyaml` runs without an explicit `actions/setup-python` step, so it targets the runner's system Python. This works today on `ubuntu-latest` (python3+pip preinstalled, and the `build` job at line 277 uses the same bare `pip install` pattern), but a future runner image with an externally-managed system Python (PEP 668) could break it with `error: externally-managed-environment`. Consider adding `actions/setup-python` before the pip installs for both jobs, or `pip install --user pyyaml`, to insulate from runner-image changes. Non-blocking — current behavior is correct.

**Review by mvillmow — COMMENTED** [review]
Re-opening 1 prior review comment(s) the current diff does not address.

File: scripts/validate_nats_config.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                break  # comment to end of line
+            if ch in "\"'":
+                instr = True
+                q = ch
+            out.append(ch)
+            i += 1
+        if instr:
+            return False, f"unterminated string on line {ln}"
+        code = "".join(out)
+        depth += code.count("{") - code.count("}")
```
Comment (mvillmow) [inline scripts/validate_nats_config.py:47]: Braces inside quoted string values are counted toward `depth`. When `instr` is true, the char (including `{`/`}`) is appended to `out`, so `code.count("{")`/`code.count("}")` on line 47 includes in-string braces. A value like `token = "a{b"` would corrupt the balance count and cause a false-positive failure. Current configs/nats/*.conf have no braces-in-strings so this doesn't bite today, but to make the heuristic robust, skip appending the char to `out` while `instr` is true (collect string contents separately, or append a placeholder space instead of the literal brace).

**Review by mvillmow — COMMENTED** [review]
GO: complete & non-vacuous NATS/compose/justfile validators in required CI job + committed pos/neg tests; one minor latent brace-in-string edge case.

**Review by mvillmow — COMMENTED** [review]
LGTM — binary-free NATS/compose validators + justfile-recipe tests correctly close #198; gates non-vacuous, exit_code symbol valid, glob covers all 6 compose files, --user pip insulates PEP 668.
